core/vm.cpp

```diff
@@ -1839,16 +1849,49 @@ class Interpreter {
                             goto replaceframe;
                         }
                     }
-                    // Equality can be used when the types don't match.
                     switch (ast.op) {
+                        // Equality can be used when the types don't match.
                         case BOP_MANIFEST_EQUAL:
                         std::cerr << "INTERNAL ERROR: Equals not desugared" << std::endl;
                         abort();
 
+                        // Equality can be used when the types don't match.
                         case BOP_MANIFEST_UNEQUAL:
                         std::cerr << "INTERNAL ERROR: Notequals not desugared" << std::endl;
                         abort();
 
+                        // e in e
+                        case BOP_IN: {
+                            if (lhs.t != Value::STRING) {
+                                throw makeError(ast.location,
+                                                "The left hand side of the 'in' operator should be "
+                                                "a string,  got " + type_str(lhs));
+                            }
+                            auto *field = static_cast<HeapString*>(lhs.v.h);
+                            switch (rhs.t) {
+                                case Value::ARRAY: {
+                                    auto *arr = static_cast<HeapArray*>(rhs.v.h);
+                                    // TODO(dcunnin): Implement this.
```

Comment:
Yeah I remember you making that argument before.  I wouldn't consider having `x in [...]` except that we already have it in the array and object comprehensions (as they're the same syntax as Python).  In that case it's obviously O(n) though.

Another complication is we're representing sets as arrays (sets are a library that makes sorted arrays).  In that case `e in my_set` would be O(n) instead of O(log n) which is even more confusing.

That could be fixed by having sets as a first class value, which can be done backwards-compatible as long as `std.type(my_set)` is "array".  However that may be even more confusing :)

I think there's enough doubt to keep `e in arr` out for the moment at least.